### PR TITLE
Remove old deprecation API

### PR DESCRIPTION
**Pull request by blindpirate**
Let's see if anything breaks.

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle test this

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/DeprecationCreationPerformanceTest.groovy [hunk 1]
```diff
@@ -24,7 +24,7 @@ class DeprecationCreationPerformanceTest extends AbstractCrossVersionPerformance
         runner.testProject = "generateLotsOfDeprecationWarnings"
         runner.tasksToRun = ['help']
         runner.gradleOpts = ["-Xms1g", "-Xmx1g"]
-        runner.minimumBaseVersion = '4.9'
+        runner.minimumBaseVersion = '6.0'
```
Comment (wolfs) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/DeprecationCreationPerformanceTest.groovy:27]: I think the `mininumBaseVersion` should be `6.3`.

**Review by wolfs — CHANGES_REQUESTED** [review]

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/DeprecationCreationPerformanceTest.groovy (hunk 1 above)
Comment (blindpirate) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/DeprecationCreationPerformanceTest.groovy:27]: Hmmm...you're righ.

Comment (blindpirate) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/DeprecationCreationPerformanceTest.groovy:27]:
```suggestion
        runner.minimumBaseVersion = '6.3'
```

**Comment by blindpirate** [comment]
@bot-gradle  test this

**Review by wolfs — APPROVED** [review]
LGTM!

**Pull request merged**